src/util/amounts.ts

```diff
@@ -0,0 +1,91 @@
+import BN from "bn.js";
+import { Fraction } from "../fraction";
+
+const MAX128 = new BN(2).pow(new BN(128)).subn(1);
+
+type SellAmount = BN;
+type BuyAmount = BN;
+
+/**
+ * Modifies the price to work with ERC20 units
+ * @param price - Amount of buy token in exchange for one sell token
+ * @param sellTokenDecimals - Number of decimals of the sell token
+ * @param buyTokenDecimals - Number of decimals of the buy token
+ * Returns Fraction representing the number of buy tokens in exchange for one unit of sell token
+ */
+export function getUnitPrice(
+  price: number,
+  sellTokenDecimals: number,
+  buyTokenDecimals: number,
+): Fraction {
+  assert(sellTokenDecimals > 0, "sell token decimals must be non-negative");
+  assert(buyTokenDecimals > 0, "buy token decimals must be non-negative");
```

Comment:
Zero decimals should be fine.
```suggestion
  assert(sellTokenDecimals >= 0, "sell token decimals must be non-negative");
  assert(buyTokenDecimals >= 0, "buy token decimals must be non-negative");
```

Reply:
I'm still hesitating about whether I should even include these assertions. If we do, then I will have to test them.

Reply:
I think assertions are nice to have in the code to demonstrate what are the developer's assumptions. I don't think you necessarily need to test them.